dask_expr/_merge.py

```diff
@@ -428,9 +428,12 @@ def _layer(self) -> dict:
         from distributed.shuffle._shuffle import shuffle_barrier
 
         dsk = {}
-        token = self._name.split("-")[-1]
-        token_left = token + "-left"
-        token_right = token + "-right"
+        token_left = _tokenize_deterministic(
+            self.left._name, self.shuffle_left_on, self.npartitions, self._partitions
```

Comment:
I assume how doesn't matter in the shuffle algorithm?

Reply:
Yes, that will only become relevant in the actual merge, which happens _after_ we shuffled all the data.